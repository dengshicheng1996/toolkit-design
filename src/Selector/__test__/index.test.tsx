import * as React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import { Selector } from '..'

describe('Selector', () => {
  it('snapshot', () => {
    const component = shallow(
      <Selector
        defaultValue={[1]}
        items={[
          {
            text: '选项一',
            value: 1,
          },
          {
            text: '选项二',
            value: 2,
          },
        ]}
      />,
    )

    const component2 = shallow(
      <Selector
        defaultValue={[1]}
        items={[
          {
            text: '选项一',
            value: 1,
          },
          {
            text: '选项二',
            value: 2,
            disabled: true,
          },
        ]}
      />,
    )

    expect(toJSON(component)).toMatchSnapshot()
    expect(toJSON(component2)).toMatchSnapshot()
  })

  it('active item className', () => {
    const component = shallow(
      <Selector
        activeItemClassName="custom-active"
        defaultValue={[1]}
        items={[
          {
            text: '选项一',
            value: 1,
          },
          {
            text: '选项二',
            value: 2,
          },
        ]}
      />,
    )

    expect(
      component
        .find('.amd-selector-item')
        .first()
        .hasClass('custom-active'),
    ).toBeTruthy()
  })
})
