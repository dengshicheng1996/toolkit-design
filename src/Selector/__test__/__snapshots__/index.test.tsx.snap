// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selector snapshot 1`] = `
<div
  className="amd-selector amd-selector-col-2"
  role="Selector"
>
  <Touchable
    key="0"
    onPress={[Function]}
  >
    <div
      className="amd-selector-item amd-selector-item-active "
    >
      <div
        className="amd-selector-item-text"
      >
        选项一
      </div>
      <div
        className="amd-selector-item-badge-active"
      />
      <Memo([object Object])
        checked={true}
        value="1"
      />
    </div>
  </Touchable>
  <Touchable
    key="1"
    onPress={[Function]}
  >
    <div
      className="amd-selector-item"
    >
      <div
        className="amd-selector-item-text"
      >
        选项二
      </div>
      <div
        className="amd-selector-item-badge-active"
      />
      <Memo([object Object])
        checked={false}
        value="2"
      />
    </div>
  </Touchable>
</div>
`;

exports[`Selector snapshot 2`] = `
<div
  className="amd-selector amd-selector-col-2"
  role="Selector"
>
  <Touchable
    key="0"
    onPress={[Function]}
  >
    <div
      className="amd-selector-item amd-selector-item-active "
    >
      <div
        className="amd-selector-item-text"
      >
        选项一
      </div>
      <div
        className="amd-selector-item-badge-active"
      />
      <Memo([object Object])
        checked={true}
        value="1"
      />
    </div>
  </Touchable>
  <Touchable
    disabled={true}
    key="1"
    onPress={[Function]}
  >
    <div
      aria-disabled={true}
      className="amd-selector-item amd-selector-item-disabled"
    >
      <div
        className="amd-selector-item-text"
      >
        选项二
      </div>
      <div
        className="amd-selector-item-badge-active"
      />
      <Memo([object Object])
        checked={false}
        disabled={true}
        value="2"
      />
    </div>
  </Touchable>
</div>
`;
